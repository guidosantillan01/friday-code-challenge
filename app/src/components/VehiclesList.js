import React from 'react';

import VehicleItem from './VehicleItem';

const VehiclesList = ({ vehicles = [] }) => {
  return (
    <div className="v-list__container">
      {vehicles.slice(0, 10).map(vehicle => {
        return <VehicleItem vehicle={vehicle} />;
      })}
    </div>
  );
};

export default VehiclesList;
