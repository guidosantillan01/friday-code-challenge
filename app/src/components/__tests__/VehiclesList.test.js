import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';

import VehiclesList from '../VehiclesList';

const vehicles = [
  {
    make: 'Tesla',
    model: 'Model S',
    enginePowerPS: '184',
    enginePowerKW: '135',
    fuelType: 'Electric',
    bodyType: 'Sedan',
    engineCapacity: '---'
  },
  {
    make: 'Tesla',
    model: 'Model 3',
    enginePowerPS: '174',
    enginePowerKW: '120',
    fuelType: 'Electric',
    bodyType: 'Sedan',
    engineCapacity: '---'
  },
  {
    make: 'Tesla',
    model: 'Model X',
    enginePowerPS: '200',
    enginePowerKW: '195',
    fuelType: 'Electric',
    bodyType: 'SUV',
    engineCapacity: '---'
  }
];

describe('Vehicle List', () => {
  afterEach(cleanup);

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<VehiclesList vehicles={vehicles} message={true} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('should render VehiclesList correctly', () => {
    const { container } = render(
      <VehiclesList vehicles={vehicles} message={true} />
    );
    expect(container).toMatchSnapshot();
  });

  test('should render the correct number of VehicleItem components', () => {
    const { getByText } = render(
      <VehiclesList vehicles={vehicles} message={true} />
    );
    expect(getByText('Tesla Model S')).toBeInTheDocument();
    expect(getByText('Tesla Model 3')).toBeInTheDocument();
    expect(getByText('Tesla Model X')).toBeInTheDocument();
  });
});
