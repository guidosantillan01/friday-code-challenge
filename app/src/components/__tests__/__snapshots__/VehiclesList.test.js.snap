// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicle List should render VehiclesList correctly 1`] = `
<div>
  <div
    class="v-list__container"
  >
    <div
      class="v-item__container"
    >
      <div>
        <img
          alt="auto"
          class="v-item__image"
          src="auto.png"
          title="auto"
        />
      </div>
      <div
        class="v-item__field"
      >
        <h3
          class="v-item__title"
          data-testid="vehicle-model"
        >
          Tesla
           
          Model S
        </h3>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Engine Power
           
          <img
            alt="power icon"
            class="v-item__icon"
            src="power.png"
          />
        </h4>
        <div
          class="v-item__engine-power"
          data-testid="vehicle-power"
        >
          <p
            class="v-item__text"
          >
            <span
              class="v-item__value"
            >
              184
            </span>
             
            <span
              class="v-item__unit"
            >
              HP
            </span>
          </p>
          <p
            class="v-item__text"
          >
            <span
              class="v-item__value"
            >
              135
            </span>
             
            <span
              class="v-item__unit"
            >
              KW
            </span>
          </p>
        </div>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Fuel Type
           
          <img
            alt="fuel icon"
            class="v-item__icon"
            src="fuel.png"
          />
        </h4>
        <p
          class="v-item__text"
          data-testid="vehicle-fuel"
        >
          Electric
        </p>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Vehicle Body Type
           
          <img
            alt="body icon"
            class="v-item__icon"
            src="body.png"
          />
        </h4>
        <p
          class="v-item__text"
          data-testid="vehicle-body"
        >
          Sedan
        </p>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Engine Capacity
           
          <img
            alt="capacity icon"
            class="v-item__icon"
            src="capacity.png"
          />
        </h4>
        <p
          class="v-item__text"
          data-testid="vehicle-capacity"
        >
          <span
            class="v-item__value"
          >
            ---
          </span>
           
          <span
            class="v-item__unit"
          >
            cc
          </span>
        </p>
      </div>
    </div>
    <div
      class="v-item__container"
    >
      <div>
        <img
          alt="auto"
          class="v-item__image"
          src="auto.png"
          title="auto"
        />
      </div>
      <div
        class="v-item__field"
      >
        <h3
          class="v-item__title"
          data-testid="vehicle-model"
        >
          Tesla
           
          Model 3
        </h3>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Engine Power
           
          <img
            alt="power icon"
            class="v-item__icon"
            src="power.png"
          />
        </h4>
        <div
          class="v-item__engine-power"
          data-testid="vehicle-power"
        >
          <p
            class="v-item__text"
          >
            <span
              class="v-item__value"
            >
              174
            </span>
             
            <span
              class="v-item__unit"
            >
              HP
            </span>
          </p>
          <p
            class="v-item__text"
          >
            <span
              class="v-item__value"
            >
              120
            </span>
             
            <span
              class="v-item__unit"
            >
              KW
            </span>
          </p>
        </div>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Fuel Type
           
          <img
            alt="fuel icon"
            class="v-item__icon"
            src="fuel.png"
          />
        </h4>
        <p
          class="v-item__text"
          data-testid="vehicle-fuel"
        >
          Electric
        </p>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Vehicle Body Type
           
          <img
            alt="body icon"
            class="v-item__icon"
            src="body.png"
          />
        </h4>
        <p
          class="v-item__text"
          data-testid="vehicle-body"
        >
          Sedan
        </p>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Engine Capacity
           
          <img
            alt="capacity icon"
            class="v-item__icon"
            src="capacity.png"
          />
        </h4>
        <p
          class="v-item__text"
          data-testid="vehicle-capacity"
        >
          <span
            class="v-item__value"
          >
            ---
          </span>
           
          <span
            class="v-item__unit"
          >
            cc
          </span>
        </p>
      </div>
    </div>
    <div
      class="v-item__container"
    >
      <div>
        <img
          alt="auto"
          class="v-item__image"
          src="auto.png"
          title="auto"
        />
      </div>
      <div
        class="v-item__field"
      >
        <h3
          class="v-item__title"
          data-testid="vehicle-model"
        >
          Tesla
           
          Model X
        </h3>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Engine Power
           
          <img
            alt="power icon"
            class="v-item__icon"
            src="power.png"
          />
        </h4>
        <div
          class="v-item__engine-power"
          data-testid="vehicle-power"
        >
          <p
            class="v-item__text"
          >
            <span
              class="v-item__value"
            >
              200
            </span>
             
            <span
              class="v-item__unit"
            >
              HP
            </span>
          </p>
          <p
            class="v-item__text"
          >
            <span
              class="v-item__value"
            >
              195
            </span>
             
            <span
              class="v-item__unit"
            >
              KW
            </span>
          </p>
        </div>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Fuel Type
           
          <img
            alt="fuel icon"
            class="v-item__icon"
            src="fuel.png"
          />
        </h4>
        <p
          class="v-item__text"
          data-testid="vehicle-fuel"
        >
          Electric
        </p>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Vehicle Body Type
           
          <img
            alt="body icon"
            class="v-item__icon"
            src="body.png"
          />
        </h4>
        <p
          class="v-item__text"
          data-testid="vehicle-body"
        >
          SUV
        </p>
      </div>
      <div
        class="v-item__field"
      >
        <h4
          class="v-item__subtitle"
        >
          Engine Capacity
           
          <img
            alt="capacity icon"
            class="v-item__icon"
            src="capacity.png"
          />
        </h4>
        <p
          class="v-item__text"
          data-testid="vehicle-capacity"
        >
          <span
            class="v-item__value"
          >
            ---
          </span>
           
          <span
            class="v-item__unit"
          >
            cc
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
